src/main/java/com/hms_networks/sc/cumulocity/CConnectorMain.java

```diff
@@ -1,24 +1,620 @@
+
+  /**
+   * The prefix appended to the front of the Ewon Flexy's serial number to form the MQTT client ID.
+   */
+  private static final String MQTT_CLIENT_ID_SERIAL_PREFIX = "HMS-Flexy-";
+
+  /** MQTT manager */
+  private static CConnectorMqttMgr mqttMgr;
+
+  /** Alarm manager */
+  private static CConnectorAlarmMgr alarmMgr;
+
+  /** Connector configuration object */
+  private static CConnectorConfigFile connectorConfig;
+
+  /** Connector control tag object for accessing its value. */
+  private static TagControl connectorControlTag = null;
+
+  /** Boolean flag to indicate if the connector is running. */
+  private static boolean isRunning = true;
+
+  /**
+   * Serial number of the host Ewon device, populated during the initialization of the connector.
+   */
+  private static String ewonSerialNumber = "";
+
+  /** Boolean indicating if the device should be restarted after the connector shuts down. */
+  private static boolean restartDeviceAfterShutdown = false;
+
+  /**
+   * Gets the connector MQTT manager.
+   *
+   * @return connector MQTT manager
+   */
+  public static CConnectorMqttMgr getMqttMgr() {
+    return mqttMgr;
+  }
+
+  /**
+   * Gets the connector configuration object.
+   *
+   * @return connector configuration
+   */
+  public static CConnectorConfigFile getConnectorConfig() {
+    return connectorConfig;
+  }
+
+  /** Triggers a shutdown of the connector and restart of the host device when requested. */
+  public static void shutdownConnectorAndRestartDevice() {
+    Logger.LOG_CRITICAL("The connector has been requested to shut down and restart the device...");
+    restartDeviceAfterShutdown = true;
+    isRunning = false;
+  }
+
+  /** Method for performing connector application initialization steps. */
+  private static boolean initialize() {
+    Logger.LOG_CRITICAL("Initializing " + CONNECTOR_FRIENDLY_NAME + "...");
+    boolean initializeSuccess = true;
+
+    // Load Ewon serial number
+    try {
+      ewonSerialNumber = new SysControlBlock(SysControlBlock.INF).getItem("SERNUM");
+    } catch (Exception e) {
+      Logger.LOG_CRITICAL("Failed to load device serial number!");
+      Logger.LOG_EXCEPTION(e);
+      initializeSuccess = false;
+    }
+
+    // Load configuration file
+    initializeSuccess &= loadConfiguration();
+
+    // Start thread for default event manager
+    boolean autorun = false;
+    EventHandlerThread eventHandler = new EventHandlerThread(autorun);
+    eventHandler.runEventManagerInThread();
+
+    // Load connector log level (default of TRACE, but should never encounter this)
+    int connectorLogLevel = Logger.LOG_LEVEL_TRACE;
+    try {
+      connectorLogLevel = connectorConfig.getConnectorLogLevel();
+    } catch (JSONException e) {
+      Logger.LOG_CRITICAL("Unable to load the connector configuration file!");
+      Logger.LOG_EXCEPTION(e);
+      initializeSuccess = false;
+    }
+    Logger.SET_LOG_LEVEL(connectorLogLevel);
```

Comment:
I don't think we do any checking on the loglevel value from the config, e.g., level=19 . Do you think this belongs here?